docs(community): Add Enterprise Interest Group charter (#2626)
modelcontextprotocol/modelcontextprotocol: docs/community/enterprise-ig/charter.mdx

---
title: Enterprise Interest Group Charter
description: Charter for the MCP Enterprise Interest Group.
---

## Group Type

**Interest Group**

## Mission Statement

> The Enterprise IG explores the challenges of deploying MCP in enterprise environments under the broader umbrella of enterprise MCP governance and best practices. As MCP adoption scales across regulated industries and large organizations, enterprise teams face recurring gaps in areas such as authentication integration, identity propagation, audit and compliance, gateway behavior, scalability and resilience, and configuration management that are not yet addressed at the protocol level. This group brings together enterprise practitioners to document these challenges, share production patterns, and build consensus on priorities for the MCP roadmap.

Reviewer: We should look at the opportunities as well.

Author: Good point @schlpbch, and agreed. Quick note that this comment is on the pre-edit version, the Mission was tightened after your earlier scope question, so the current charter already leads with surfacing enterprise requirements gaps and channeling them to the WGs rather than a best-practices catalog.

On opportunities: I'd frame it as the flip side of the same mandate. Many of these gaps, once closed, are exactly what unlocks enterprise value and adoption, so the catalog can capture the opportunity each requirement represents, not just the pain. I'll make sure that framing is explicit. I'd keep it tied to the requirements-to-WG flow though, so we capture the upside without the scope drifting back into open-ended territory. What do you think?

Author: Added this to the charter, the overarching In-Scope item now states each requirement is captured with the enterprise opportunity it represents, not just the pain point, while keeping it tied to the requirements-to-WG flow. Thanks, this made it sharper.

> The Enterprise IG identifies and documents the requirements gaps that enterprises encounter when deploying MCP in production. As MCP adoption scales across regulated industries and large organizations, enterprise teams face recurring gaps in areas such as authentication integration, identity propagation, audit and compliance, gateway behavior, scalability and resilience, and configuration management that are not yet addressed at the protocol level. The group brings together enterprise practitioners to capture these gaps as problem statements and structured recommendations, and to channel them to the relevant Working Groups for specification work. Governance and best-practice patterns may emerge from these discussions, but the group’s core mandate is surfacing protocol-level requirements, not maintaining a general best-practices catalog.

Reviewer: ```suggestion
The Enterprise IG identifies and documents the requirements gaps that enterprises encounter when deploying MCP in production. As MCP adoption scales across regulated industries and large organizations, enterprise teams face recurring gaps in areas such as authentication integration, identity propagation, audit and compliance, gateway behavior, scalability and resilience, and configuration management that are not yet addressed at the protocol level. 

The group brings together enterprise practitioners to capture these gaps as problem statements and structured recommendations, and to channel them to the relevant Working Groups for specification work. Governance and best-practice patterns may emerge from these discussions, but the group’s core mandate is surfacing protocol-level requirements, not maintaining a general best-practices catalog.
```

Reviewer: splitting into two paragraphs

Author: Done, split the Mission into two paragraphs as suggested. Reads better. Thanks.

All discussions and sessions are protocol-focused and vendor-neutral. The Enterprise IG is not a venue for product pitches or vendor-specific marketing. Participants contribute as practitioners sharing deployment experience and requirements, not as representatives promoting commercial solutions.

## Scope

### In Scope

- **Enterprise Requirements Gaps (overarching focus)**: The group’s primary mandate is identifying enterprise deployment requirements that are not addressed at the MCP protocol level today and turning them into problem statements and recommendations for the relevant Working Groups. The specific domains below are explored through that lens; governance and best-practice patterns are documented only where they surface a protocol-level gap:
- **Enterprise Authentication and Identity**: Discussion of enterprise-managed auth patterns including IdP integration (SSO, SAML, OIDC), token lifecycle management, On-Behalf-Of (OBO) token exchange flows, and fine-grained authorization (Rich Authorization Requests). Gathering requirements to inform the Auth WG and related SEPs.

> - **Identity Propagation and Session Context**: How verified identity is surfaced and conveyed across MCP clients, servers, and downstream services for policy enforcement and audit. Includes the spawned-agent and delegated-agent case: identity lineage, least-privilege for child agents, descendant revocation, and auditability across multi-agent chains. Adjacent to authentication but distinct in focus — the gaps show up in how identity is exposed and consumed, not how auth itself is performed.

Reviewer: ```suggestion
- **Identity Propagation and Session Context**:  Defines how  averified identity is surfaced and conveyed across MCP clients, servers, and downstream services for policy enforcement and audit.  This includes spawned-agent and delegated-agent use cases: identity lineage, least-privilege for child agents, descendant revocation, and auditability across multi-agent chains which is adjacent to authentication but distinct in focus.
```

Author: Took this rephrase, reads cleaner. Thanks.

- **Audit, Observability, and Compliance**: Exploration of standardized audit trail formats for MCP tool invocations, integration with enterprise SIEM and observability platforms, and evidence generation for compliance frameworks (SOC 2, HIPAA, GDPR, EU AI Act). Includes sector-specific compliance patterns such as PHI handling and BAA-scoped data flows in healthcare environments.

> - **Gateway and Proxy Behavior**: Documentation of enterprise deployment patterns where MCP traffic routes through API gateways, load balancers, reverse proxies, and security proxies. Identification of protocol-level gaps (session stickiness, SSE timeout handling, header propagation) that affect enterprise deployments.

Reviewer: The next version of MCP is stateless, thus session stickiness is solved IMHO.

Author: Good catch @schlpbch , you're right. With the move to stateless streamable HTTP, session stickiness isn't a meaningful gap, so it shouldn't be in the list. I'll drop it (and reconsider the SSE example for the same reason) and keep the gateway scope to things that still hold in a stateless model, header propagation, connection lifecycle, load balancing across stateless connections.

Author: Done in the latest push, dropped session stickiness and SSE, and the gateway gaps now focus on what persists in a stateless model (header/context propagation, authorization handoff across proxies, policy enforcement at the gateway). Thanks for catching this.

> - **Scalability and Resilience**: Patterns for running MCP at enterprise scale across regions and availability zones, handling failover, managing session state under horizontal scaling, and ensuring reliability for mission-critical deployments.

Reviewer: Statlessness allows horizontal scaling.

Reviewer: I'd put patterns/best practices out of scope.

Author: Both addressed. Reframed Scalability around the stateless core, with horizontal scaling now enabled without sticky sessions, and dropped session stickiness from the gateway gaps for the same reason. On patterns/best practices: tightened the language throughout so the In-Scope intro now states governance and best-practice patterns are documented only where they surface a protocol-level gap, and Out of Scope explicitly excludes a general best-practices catalog. The mandate stays on requirements feeding the WGs. Thanks, this sharpened it.

> - **MCP in Enterprise Architecture**: How MCP fits into broader enterprise architecture including integration with existing middleware, data platforms, and agent orchestration layers.

Reviewer: How to transform any protocol/technology to MCP and back should not be part of a standard, scope becomes huge in that case.

> - **MCP in Enterprise Architecture**: How MCP fits into broader enterprise architecture including integration with existing middleware, data platforms, and agent orchestration layers.

Reviewer: ```suggestion
- **MCP in Enterprise Architecture**:  Definition of how MCP fits into broader enterprise architecture including integration with existing middle-ware, data platforms, and agent orchestration layers.
```

Author: Took the "Definition of" rephrase. Kept "middleware" unhyphenated since that's the standard spelling. Thanks.

- **Configuration Portability and Deployment**: Discussion of challenges in deploying MCP servers across heterogeneous client environments, multi-tenant configuration patterns, and enterprise-scale rollout practices.
- **Interceptors and Middleware**: Exploration of use cases for request/response interception in MCP pipelines, including PII redaction, compliance validation, content filtering, and hallucination detection. Gathering requirements that may inform future specification work on middleware patterns.
- **Problem Statements and Use Cases**: Documentation of enterprise pain points and production failure scenarios (anonymized) to build an evidence base for prioritizing specification work.
- **Recommendations to Working Groups**: Providing structured input from enterprise practitioners to the Auth WG, Security IG, Gateways IG, and other groups on enterprise-specific requirements.

### Out of Scope

- Writing or owning SEPs. The Enterprise IG produces problem statements, use cases, and recommendations. Concrete specification changes are driven through the appropriate Working Groups.

> - Defining product-specific or vendor-specific solutions. Discussions must remain vendor-neutral and focused on protocol-level patterns.

Reviewer: ```suggestion
- Discussion of product-specific or vendor-specific solutions. Discussions must remain vendor-neutral and focused on protocol-level patterns.
```

- General MCP support or onboarding. The group focuses on enterprise deployment challenges, not introductory MCP topics.
- Overlapping directly with the Auth WG's specification work. The Enterprise IG gathers enterprise auth requirements and feeds them to the Auth WG rather than proposing auth protocol changes.

### Related Groups

> - **Auth WG** — Enterprise auth requirements (IdP integration, OBO flows, RAR) are gathered here and provided as input to the Auth WG's specification work.

Reviewer: ```suggestion
#### Working Group 

- **Transport WG** — Potentials Transport level improvements needed for enterprise support (e.g. non-functional requirements such as latency or horizontal scaling) are bundled and provided as input to the Transport WG.
 
- **Extensions WG**  —  Requirements for extensions other than auhorizattion are bundeled and provided as input to the Extension WG.

- **Auth WG** — Enterprise auth requirements (IdP integration, OBO flows, RAR) are gathered here and provided as input to the Auth WG's specification work.
```

Reviewer: Added two relevant working groups

Author: Added both, Transport WG and Extensions WG are now under the Working Groups subsection. Cleaned up a couple of typos in the process. Thanks for flagging these, the WG list is more complete now.

> - **Security IG** — Overlapping interest in runtime security, tool poisoning, and credential management. The Enterprise IG focuses on the organizational and governance dimensions; the Security IG focuses on threat models and protocol-level mitigations.

Reviewer: ```suggestion

#### Interest Group
- **Auth IG** — Direct overlap with the  **Auth WG*
- **Security IG** — Overlapping interest in runtime security, tool poisoning, and credential management. 
```

Author: Adopted the Working Group / Interest Group split you suggested, and added Transport WG and Extensions WG under Working Groups. I kept Auth under Working Groups rather than adding an Auth IG, since Auth is a WG and it's already listed there, didn't want to duplicate it. Security, Gateways, and FS-IG sit under Interest Groups. Thanks, the split is cleaner.

| Raghu Chandra | Independent | [@raghu-chandra-mcp](https://github.com/raghu-chandra-mcp) | raghu.chandra | Facilitator |
| Yannj_Fr | MCPApps Builders | [@yannj-fr](https://github.com/yannj-fr) | yannj_fr | Facilitator |
| Peder H P | Saxo Bank | [@pederhp](https://github.com/pederhp) | | Participant |
| Derek Lewis | Silex Data Solutions | [@derekelewis](https://github.com/derekelewis) | dlewis.io | Participant |
| Aman s | Independent Researcher; Blue Shield of California | [@aman210122](https://github.com/aman210122) | | Participant |
| Ola | Nordstrom / MCP Maintainer | [@olaservo](https://github.com/olaservo) | | Participant |
| Varun | TraceForce | [@vawadhwa88](https://github.com/vawadhwa88) | | Participant |
| Markus Mueller | Boomi | [@mquadrat](https://github.com/mquadrat) | | Participant |
| Aaron Parecki | Okta | [@aaronpk](https://github.com/aaronpk) | | Participant |
| cayerbe | GNS-Foundation | [@GNS-Foundation](https://github.com/GNS-Foundation) | cayerbe | Participant |
| Anishma | EmpowerID | [@anishma](https://github.com/anishma) | | Participant |
| Lin Sun | Solo.io | [@linsun](https://github.com/linsun) | | Participant |

> | Joey | Archestra | [@joeyorlando](https://github.com/joeyorlando) | | Participant |

Reviewer 2: ```suggestion
| Joey Orlando | Archestra | [@joeyorlando](https://github.com/joeyorlando) | | Participant |
```

thank you for formalizing this charter doc @raghu-chandra-mcp 👏

| Working Session | Monthly | 60 min | Use case discussion, pain point cataloging, cross-group coordination |

Discord: #enterprise-ig

### Decision Process for Recommendations

The Enterprise IG produces non-binding recommendations, problem statements, and use cases. To ensure clarity and avoid downstream process debates, the following applies:

- **Problem statements and use cases** are contributed asynchronously via GitHub Discussions or Discord threads and reviewed during monthly meetings.
- **Recommendations to Working Groups** are finalized through lazy consensus among participants, with a 7-day review period posted on GitHub. If no objections are raised, the recommendation is considered approved.
- **Facilitators** are responsible for packaging approved recommendations and delivering them to the relevant Working Group or Interest Group.

## Deliverables (Optional)

As an Interest Group, the Enterprise IG does not produce binding deliverables. The following are planned discussion outputs:

| Item | Status | Target Date | Champion |
| ---- | ------ | ----------- | -------- |
| Enterprise Pain Points Catalog | Planned | Q2 2026 | Raghu Chandra |

> | Healthcare & Compliance Use Cases (PHI, BAA) | Planned | Q3 2026 | Aman s |

Reviewer: Potential extendable to "Regulated Environnements Use Cases" like Health Care, Finance, Governement, Defense just to name a few.

| Enterprise Auth Requirements (input to Auth WG) | Planned | Q3 2026 | TBD |

> | Gateway Deployment Patterns Document | Planned | Q3 2026 | TBD |

Reviewer: The following topics I see important as well
- traceability and audit ability
- non functional requirements like chattiness, speed, size, scalabilty...

> The Enterprise IG produces non-binding recommendations, problem statements, and use cases. To ensure clarity and avoid downstream process debates, the following applies:

Reviewer: ```suggestion
The Enterprise IG ideally produces use cases and functional as well as non-functional requirements. To ensure clarity and avoid downstream process debates, the following applies:
```

Author: Took your framing toward functional and non-functional requirements, just kept "non-binding" (and the problem-statements wording) for the reason in my top-level comment: it signals the IG informs the WGs rather than dictating to them, which keeps the scope boundary clean. Thanks.

All discussions and sessions are protocol-focused and vendor-neutral. The Enterprise IG is not a venue for product pitches or vendor-specific marketing. Participants contribute as practitioners sharing deployment experience and requirements, not as representatives promoting commercial solutions.

## Scope

### In Scope

- **Enterprise Requirements Gaps (overarching focus)**: The group’s primary mandate is identifying enterprise deployment requirements that are not addressed at the MCP protocol level today and turning them into problem statements and recommendations for the relevant Working Groups. Each requirement is captured with the enterprise opportunity it represents, not only the pain point, since closing these gaps is what unlocks broader enterprise adoption and value. The specific domains below are explored through that lens; governance and best-practice patterns are documented only where they surface a protocol-level gap:

> - **Enterprise Authentication and Identity**: Discussion of enterprise-managed auth patterns including IdP integration (SSO, SAML, OIDC), token lifecycle management, On-Behalf-Of (OBO) token exchange flows, and fine-grained authorization (Rich Authorization Requests). Gathering requirements to inform the Auth WG and related SEPs.

Reviewer 2: isn't this "covered" by the Enterprise Managed Auth extension of the protocol?

Reviewer: My take on that is: this group identifies whatever lacking requirements enterprises environnements have, and in case of gaps passes them to the Auth group.
